include/flatbuffers/flatbuffers.h

```diff
@@ -616,19 +640,29 @@ class vector_downward {
 
   Allocator *allocator_;
   bool own_allocator_;
+  uoffset_t initial_size_;
   size_t reserved_;
   uint8_t *buf_;
   uint8_t *cur_;  // Points at location between empty (below) and used (above).
 
   void reallocate(size_t len) {
-    size_t old_reserved = reserved_;
+    assert(allocator_ != nullptr);
+    auto old_reserved = reserved_;
     auto old_size = size();
-    auto largest_align = AlignOf<largest_scalar_t>();
-    reserved_ += (std::max)(len, growth_policy(reserved_));
-    // Round up to avoid undefined behavior from unaligned loads and stores.
-    reserved_ = (reserved_ + (largest_align - 1)) & ~(largest_align - 1);
-    buf_ = allocator_->reallocate_downward(buf_, old_reserved, reserved_);
-    cur_ = buf_ + reserved_ - old_size;
+    size_t reserved;
+    if (old_reserved == 0) {
```

Comment:
Actually now that I look at this, we should just put the `reserved_ == 0` case in `growth_policy(...)`.

`initial_size` is a `uoffset_t` because that's how it was in the original implementation. I think it could be a `size_t` across the board, so I'll go ahead and do that if we don't care about the change in size/alignment.